packages/rich-text/src/plugins/CommandPalette/useCommands.ts

```diff
@@ -0,0 +1,186 @@
+                          }
+                        },
+                      };
+                    })
+                  );
+                }
+              });
+            },
+          },
+          {
+            id: `${contentType.sys.id}-inline`,
+            label: `Embed ${contentType.name} - Inline`,
+            callback: () => {
+              fetchEntries(sdk, contentType, query).then((entries) => {
+                removeQuery(editor);
+                if (!entries.length) {
+                  setCommands([
+                    {
+                      id: 'no-results',
+                      label: 'No results',
+                    },
+                  ]);
+                } else {
+                  setCommands(
+                    entries.map((entry) => {
+                      return {
+                        id: `${entry.id}-${entry.displayTitle.replace(/\W+/g, '-').toLowerCase()}`,
+                        label: entry.displayTitle,
+                        callback: () => {
+                          const inlineNode = createInlineEntryNode(entry.id);
+                          removeCommand(editor);
+                          Transforms.insertNodes(editor, inlineNode);
+                          editor.insertText('');
+                        },
+                      };
+                    })
+                  );
+                }
+              });
+            },
+          },
+        ],
+      };
+    });
+    const assetCommand = {
+      group: 'Assets',
+      commands: [
+        {
+          id: 'embed-asset',
+          label: 'Embed Asset',
+          callback: () => {
+            fetchAssets(sdk, query).then((assets) => {
+              removeQuery(editor);
+              if (!assets.length) {
+                setCommands([
+                  {
+                    id: 'no-results',
+                    label: 'No results',
+                  },
+                ]);
+              } else {
+                setCommands(
+                  assets.map((asset) => {
+                    return {
+                      id: `${asset.id}-${asset.displayTitle.replace(/\W+/g, '-').toLowerCase()}`,
+                      label: asset.displayTitle,
+                      thumbnail: asset.thumbnail,
+                      callback: () => {
+                        removeCommand(editor);
+                        if (editor.selection) {
+                          const selection = editor.selection;
+                          editor.insertSoftBreak();
+                          insertBlock(editor, BLOCKS.EMBEDDED_ASSET, asset.entry);
+                          Transforms.select(editor, selection);
+                        }
+                      },
+                    };
+                  })
+                );
+              }
+            });
+          },
+        },
+      ],
+    };
+    return [...contentTypeCommands, assetCommand];
+  });
+
+  return query
+    ? commands.reduce((list, nextItem) => {
```

Comment:
can we leave an explanatory comment here of what this commands.reduce is doing? the logic is a bit hard to follow